File: src/tmp/Java/src/com/dbc/algorithm/ReferOffer/QueShiDeShuZiLcof.java
package com.dbc.algorithm.ReferOffer;

public class QueShiDeShuZiLcof {
    public int missingNumber(int[] nums) {
        int left = 0, right = nums.length - 1, res = nums.length;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (nums[mid] != mid) {
                res = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return res;
    }
}

File: src/tmp/Python/refer-offer/que_shi_de_shu_zi_lcof.py
from typing import *

def missingNumber(nums: List[int]) -> int:
    left, right = 0, len(nums) - 1
    res = -1
    while left <= right:
        mid = (left + right) // 2
        if nums[mid] != mid:
            res = mid
            right = mid - 1
        else:
            left = mid + 1
    
    return res

if __name__ == "__main__":
    print(missingNumber([0,1,2,3,4,5,6,7,9]))